iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/RemoveHvdcLine.java

```diff
@@ -0,0 +1,116 @@
+                        hvdcConverterStation2Atomic.set(hvdcLine.getConverterStation2());
+                        hvdcLine.remove();
+                        hvdcLineRemoved.set(true);
+                        removedHvdcLineReport(reporter, hvdcLineId);
+
+                    }
+                });
+
+        HvdcConverterStation<? extends HvdcConverterStation> hvdcConverterStation1 = hvdcConverterStation1Atomic.get();
+        HvdcConverterStation<? extends HvdcConverterStation> hvdcConverterStation2 = hvdcConverterStation2Atomic.get();
+
+        if (!hvdcLineRemoved.get()) {
+            LOGGER.error("HVDC Line {} not found", hvdcLineId);
+            notFoundHvdcLineReport(reporter, hvdcLineId);
+            if (throwException) {
+                throw new PowsyblException("HVDC Line not found: " + hvdcLineId);
+            }
+        } else {
+            removeConverterStations(hvdcConverterStation1, hvdcConverterStation2, reporter);
+
+            // Remove the MCSs that represent the filters of the LCC
+            if (mscIds != null
+                    && (isLccConverterStation(hvdcConverterStation1)
+                    || isLccConverterStation(hvdcConverterStation2))) {
+                removeMCSs(network, reporter);
+            }
+        }
+    }
+
+    private void removeConverterStations(HvdcConverterStation<? extends HvdcConverterStation> hvdcConverterStation1, HvdcConverterStation<? extends HvdcConverterStation> hvdcConverterStation2, Reporter reporter) {
+        hvdcConverterStation1.remove();
+        hvdcConverterStation2.remove();
+        reportConverterStationRemoved(reporter, hvdcConverterStation1);
+        reportConverterStationRemoved(reporter, hvdcConverterStation2);
+    }
+
+    private boolean isLccConverterStation(HvdcConverterStation<? extends HvdcConverterStation> hvdcConverterStation) {
+        Objects.requireNonNull(hvdcConverterStation);
+        return hvdcConverterStation.getHvdcType() == HvdcConverterStation.HvdcType.LCC;
+    }
+
+    private boolean isVscConverterStation(HvdcConverterStation<? extends HvdcConverterStation> hvdcConverterStation) {
+        Objects.requireNonNull(hvdcConverterStation);
+        return hvdcConverterStation.getHvdcType() == HvdcConverterStation.HvdcType.VSC;
+    }
+
+    private void reportConverterStationRemoved(Reporter reporter, HvdcConverterStation<? extends HvdcConverterStation> hvdcConverterStation) {
+        if (isLccConverterStation(hvdcConverterStation)) {
+            removedLccConverterStationReport(reporter, hvdcConverterStation.getId());
+        }
+        if (isVscConverterStation(hvdcConverterStation)) {
+            removedVscConverterStationReport(reporter, hvdcConverterStation.getId());
+        }
+    }
+
+    private void removeMCSs(Network network, Reporter reporter) {
+        mscIds.forEach(mscId ->
+                network.getShuntCompensatorStream().filter(shuntCompensator -> shuntCompensator.getId().equals(mscId))
```

Comment:
This is very inefficient!!! There is network.getShuntCompensator to get a shunt from its id. Here you iterate on all shunts and post filter while there is already inside network model index to get quickly a shunt from its ID.